Simplenote/Classes/String+Simplenote.swift

```diff
@@ -45,4 +45,125 @@ extension String {
 
         return output
     }
+
+    /// Returns a new string dropping specified prefix
+    ///
+    func droppingPrefix(_ prefix: String) -> String {
+        guard hasPrefix(prefix) else {
+            return self
+        }
+
+        return String(dropFirst(prefix.count))
+    }
+}
+
+// MARK: - Searching for the first / last characters
+//
+extension String {
+
+    /// Find and return the location of the first / last character from the specified character set
+    ///
+    func locationOfFirstCharacter(from searchSet: CharacterSet,
+                                  startingFrom startLocation: String.Index,
+                                  backwards: Bool = false) -> String.Index? {
+
+        guard startLocation <= endIndex else {
+            return nil
+        }
+
+        let range: Range<String.Index> = {
+            if backwards {
+                return startIndex..<startLocation
+            }
+
+            return startLocation..<endIndex
+        }()
```

Comment:
Hooow about:

```
let range = backwards ? startIndex..<startLocation : startLocation..<endIndex
```

(Less lines, one less context switch)

Reply:
Huh I'm not 100% sure if an inline anonymous function like this one would or would not produce a stack dump? I'd say yes, buuut I haven't checked the ASM!!

Reply:
Not sure why I went with an anonymous function here, doesn't make much sense 🤦 
Thank you for the suggestion, looks beautiful!